### PR TITLE
支持sonar 9.9生成

### PR DESCRIPTION
**Pull request by gosszhang**
1. 支持sonarqube9.9 api生成
2. 增加auth参数，可选
3. 使用golang1.20
4. 优化模板结构

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -0,0 +1,8 @@
+module example.com/m/v2
+
+go 1.20
+
+require (
+	github.com/google/go-querystring v1.1.0
+	github.com/pkg/errors v0.9.1
```
Comment (RidgeA) [inline go.mod]: Please, use built-in package instead (https://pkg.go.dev/errors), this one is abandoned.

File: go.mod [hunk 2]
```diff
@@ -0,0 +1,8 @@
+module example.com/m/v2
+
+go 1.20
+
+require (
+	github.com/google/go-querystring v1.1.0
```
Comment (RidgeA) [inline go.mod]: This package seems to be not used in the project, please remove it.

**Review by RidgeA — CHANGES_REQUESTED** [review]
I appreciate your interest in the package, I hope it will be helpful for you. 
Please use English in the future, I don't speak Chinese.

**Comment by gosszhang** [comment]
The issues you raised have been fixed. Could you please review them again for me.

**Review by RidgeA — APPROVED** [review]

**Pull request merged**